core/services/ocr2/plugins/ocr2vrf/coordinator/proxy.go

```diff
@@ -0,0 +1,82 @@
+package coordinator
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/gethwrappers/generated"
+	"github.com/smartcontractkit/chainlink/core/gethwrappers/ocr2vrf/generated/vrf_beacon"
+	"github.com/smartcontractkit/chainlink/core/gethwrappers/ocr2vrf/generated/vrf_coordinator"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+var _ VRFBeaconCoordinator = &VRFProxy{}
+
+//go:generate mockery --name VRFBeaconCoordinator --output ./mocks/ --case=underscore
+
+// VRFProxy routes requests to VRFBeacon and VRFCoordinator go wrappers and implements VRFBeaconCoordinator interface
+type VRFProxy struct {
+	lggr        logger.Logger
+	beacon      vrf_beacon.VRFBeacon
+	coordinator vrf_coordinator.VRFCoordinator
```

Comment:
It's more consistent with the rest of the codebase that these are pointers and not values.

Reply:
made these interfaces instead per your comment below.